src/transformer/descriptor/helper/helper.ts

```diff
@@ -70,6 +55,22 @@ export namespace TypescriptHelper {
         }
 
         return TypeChecker().typeToTypeNode(type);
+    }
+
+    export function getTypeParameterOwnerMock(d: ts.Declaration): ts.Declaration {
+        const typeDeclaration: ts.Declaration = ts.getTypeParameterOwner(d);
 
+        // THIS IS TO FIX A MISSING IMPLEMENTATION IN TYPESCRIPT https://github.com/microsoft/TypeScript/blob/ba5e86f1406f39e89d56d4b32fd6ff8de09a0bf3/src/compiler/utilities.ts#L5138
+        if ((typeDeclaration as Declaration).typeParameters) {
+            return typeDeclaration;
+        }
+
+        const parent: ts.Node = d.parent;
```

Comment:
this code can be refactored a bit
```
for(let current: ts.Node = d; current = current.parent;) {
  if (current.kind === ts.SyntaxKind.TypeAliasDeclaration) {
    return current as ts.Declaration;
  }
}
```

Reply:
I've copied this from the typescript implementation.
https://github.com/microsoft/TypeScript/blob/ba5e86f1406f39e89d56d4b32fd6ff8de09a0bf3/src/compiler/utilities.ts#L5138

 I've tried to remove the code you've wrote but apparently I have to specify also "current;" before the new value each loop.

specifying that value will evaluate every time the expression, so if current is not define it will stop iterate through it and return undefined.

Reply:
it's ok I'm not overly attached to my alternative... but I'm pretty sure it works, it's strange that it doesn't work for you

closing this comment